src/main/java/seedu/academydirectory/model/student/Attendance.java

```diff
@@ -0,0 +1,76 @@
+package seedu.academydirectory.model.student;
+
+import static java.util.Objects.requireNonNull;
+
+public class Attendance {
+
+    private boolean[] attendanceInBoolean;
+
+    /**
+     * The constructor for an Attendance object.
+     * @param numberOfSessions The number of Studios scheduled
+     *                         for the semester that the student has to attend
+     */
+    public Attendance(Integer numberOfSessions) {
+        requireNonNull(numberOfSessions);
+        attendanceInBoolean = new boolean[numberOfSessions];
+    }
+
+    /**
+     * Edits the attendance of a student in the specified Studio session.
+     * @param index The session number
+     * @param attendanceStatus true if attended, false otherwise
+     */
+    public Attendance update(Integer index, boolean attendanceStatus) {
+        attendanceInBoolean[index - 1] = attendanceStatus;
+        return this;
+    }
+
+    public void setAttendance(boolean[] boolArr) {
+        this.attendanceInBoolean = boolArr;
+    }
+
+    public Integer getSessionCount() {
+        return attendanceInBoolean.length;
+    }
+
+    public boolean[] getAttendanceInBoolean() {
+        return this.attendanceInBoolean;
+    }
+
+    @Override
+    public String toString() {
+        String str = "";
+        for (int i = 0; i < attendanceInBoolean.length; i++) {
+            str += attendanceInBoolean[i] ? "[" + (i + 1) + "]" : "[ ]";
+        }
+        return str;
+    }
```

Comment:
This is a bit complicated to be honest. Try to make the code a bit more explicit - using branch statement outright rather than shortened conditional expression.